drivers/sensor/max30208/max30208.h

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2021, Silvano Cortesi
+ * Copyright (c) 2021, ETH Zürich
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_MAX30208_H_
+#define ZEPHYR_DRIVERS_SENSOR_MAX30208_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <drivers/sensor.h>
+#include <drivers/sensor/max30208.h>
+#include <drivers/i2c.h>
+#include <drivers/gpio.h>
+
+/* Macros for struct to byte conversion */
+#define MAX30208_STC_TO_P(STRUCT) (uint8_t *)&(STRUCT)
+#define MAX30208_STC_TO_B(STRUCT) *MAX30208_STC_TO_P(STRUCT)
+
+/* Interrupt and Status */
+#define MAX30208_REG_INT_STS		0x00
+#define MAX30208_REG_INT_EN		0x01
+
+/* FIFO */
+#define MAX30208_REG_FIFO_WR		0x04
+#define MAX30208_REG_FIFO_RD		0x05
+#define MAX30208_REG_FIFO_OVF		0x06
+#define MAX30208_REG_FIFO_DATA_CTR	0x07
+#define MAX30208_REG_FIFO_DATA		0x08
+#define MAX30208_REG_FIFO_CFG1		0x09
+#define MAX30208_REG_FIFO_CFG2		0x0A
+
+/* System */
+#define MAX30208_REG_SYS_CTRL		0x0C
+
+/* Temperature */
+#define MAX30208_REG_ALRM_H_MSB		0x10
+#define MAX30208_REG_ALRM_H_LSB		0x11
+#define MAX30208_REG_ALRM_L_MSB		0x12
+#define MAX30208_REG_ALRM_L_LSB		0x13
+#define MAX30208_REG_TEMP_SETUP		0x14
+
+/* GPIO */
+#define MAX30208_REG_GPIO_SETUP		0x20
+#define MAX30208_REG_GPIO_CTRL		0x21
+
+/* Identifiers */
+#define MAX30208_REG_PART_ID1		0x31
+#define MAX30208_REG_PART_ID2		0x32
+#define MAX30208_REG_PART_ID3		0x33
+#define MAX30208_REG_PART_ID4		0x34
+#define MAX30208_REG_PART_ID5		0x35
+#define MAX30208_REG_PART_ID6		0x36
+#define MAX30208_REG_PART_ID		0xFF
+
+#define MAX30208_PART_ID		0x30
+
+/* defines for conversion */
+#define MAX30208_ONE_DEGREE		0xC8
+#define MAX30208_LSB_e6			5000
+
+/* enums and definitions */
+#define MAX30208_RESET_MASK		0x01
+#define MAX30208_CONVERT_T_MASK		0xFF
+#define MAX30208_INT_A_FULL_MASK	(1 << 7)
+#define MAX30208_INT_TEMP_LO_MASK	(1 << 2)
+#define MAX30208_INT_TEMP_HI_MASK	(1 << 1)
+#define MAX30208_INT_TEMP_RDY_MASK	(1 << 0)
+#define MAX30208_GPIO_MODE_MASK(gpioX, mode) (mode << (gpioX * 6))
+#define MAX30208_POLL_TRIES		10
+#define MAX30208_POLL_TIME		10
+#define MAX30208_TMP_MEAS_TIME		10
+
+enum MAX30208_gpio_mode {
+	MAX30208_GPIO_INPUT = 0,
+	MAX30208_GPIO_OUTPUT = 1,
+	MAX30208_GPIO_INPUT_PULLDOWN = 2,
+	MAX30208_GPIO_CONV_TEMP = 3,
+	MAX30208_GPIO_INTB = 3,
+};
+
+/* register structs */
+#ifdef CONFIG_BIG_ENDIAN
+struct MAX30208_status {
```

Comment:
naming convention:
```suggestion
struct max30208_status {
```

Reply:
what for enums? also lower case for its name (and then inside upper case)?